vision/grid_map/grid_map.py

```diff
@@ -0,0 +1,271 @@
+                    s2.delete = 0
+                    break
+
+    # If a square is not visible for 20 consecutive frames, delete it
+    for s in squares:
+        if not s.visible:
+            if s.delete == 20:
+                squares.remove(s)
+            else:
+                s.delete += 1
+
+    if len(curr_squares) != len(matched):
+        for s in curr_squares:
+            if s.id not in matched:
+                id += 1
+                s.id = id
+                matched.append(s.id)
+                squares.append(s)
+
+    screenshot_squares(squares, frame)
+
+    # Drawing squares on the frame
+    for s in squares:
+        if s.visible:
+            cv2.putText(frame, str(s.id), (s.x + int(s.w / 2), s.y + int(s.w / 2)),
+                        cv2.FONT_HERSHEY_DUPLEX, 1, (0, 255, 0))
+            cv2.rectangle(frame, (s.x, s.y), (s.x + s.w, s.y + s.h), (0, 255, 0), 5)
+
+def screenshot_squares(squares, frame):
+    for s in squares:
+        if s.screenshot_num != 5:
+            s.screenshot_num += 1
+
+            roi = frame[s.y:s.y + s.h, s.x:s.x + s.w]
+            file_name = str(s.id) + "(" + str(s.screenshot_num) + ").jpg"
+            cv2.imwrite(get_folder("transect", "squares") + "/" + file_name, roi)
+
+def display_grid():
+    cell_size = 150
+    padding = 8 # Line thickness
+    border = 100 # Border around grid
+
+    width = 3 * cell_size + 4 * padding  # width of window
+    height = 9 * cell_size + 10 * padding  # height of window
+
+    img = np.zeros((height + 2 * border, width + 2 * border, 3), dtype=np.uint8)
+    img.fill(255)
+
+    start_x = border + padding
+    start_y = border + padding
+
+    # Draw vertical griid lines
+    for i in range(4):
+        cv2.line(img, (start_x, start_y), (start_x, height + border), (0, 0, 0), padding, 1)
+        start_x += cell_size + padding
+
+    start_x = border + padding # Reset x position
+
+    # Draw horizontal grid lines
+    for i in range(10):
+        cv2.line(img, (start_x, start_y), (width + border, start_y), (0, 0, 0), padding, 1)
+        start_y += cell_size + padding
+
+    # Insert text
+    text = "Side of pool"
+    font = cv2.FONT_HERSHEY_SIMPLEX
+
+    text_size = cv2.getTextSize(text, font, 2, 2)[0]
+    text_x = int((img.shape[1] - text_size[0]) / 2)
+    text_y = int(border + height + border/1.5)
+
+    cv2.putText(img, text, (text_x, text_y), font, 2, (0, 0, 0))
+    show_debug(img, name="frame", wait=True)
+
+
+# Used for trackbar
+def empty(a):
+    pass
+
+def start_mapping(video):
+
+    cv2.namedWindow("Trackbar")
+    cv2.createTrackbar("Thresh1", "Trackbar", 87, 255, empty)
+    cv2.createTrackbar("Thresh2", "Trackbar", 230, 255, empty)
+
+    clear_folder("transect", "squares")
+
+    while video.isOpened():
+        ret, frame = video.read()
+
+        if not ret:
+            break
+
+        blur = cv2.GaussianBlur(frame, (7, 7), 1)
+
+        thresh1 = cv2.getTrackbarPos("Thresh1", "Trackbar")
```

Comment:
Since your algorithm has a lot of different parameters, it would be a good idea to put them all in one place that's easy to adjust. These could be fields in the class you make. Then in the testing script, you can add sliders that will adjust the values of the class fields.